web/packages/teleport/src/lib/term/terminal.ts

```diff
@@ -78,6 +80,12 @@ export default class TtyTerminal {
 
     this.term.loadAddon(this._fitAddon);
     this.term.loadAddon(this._webLinksAddon);
+    // handle context loss and load webgl addon
+    this._webglAddon.onContextLoss(() => {
+      this._webglAddon.dispose();
```

Comment:
i think we also need to also call this in the `destory()` func.

i'm not sure i understand what the `onContextLoss` does, but shouldn't we try to re-instantiate it after disposing?

Reply:
Yeah we need it in `destroy()` for sure. thank you!

`onContextLoss` is essentially when the browser loses access to the GPU when rendering (if you do some weird driver updates or some kind of switch between GPUs). it's rare. However, the addon itself runs a timeout internally to automatically reinstantiate after a couple seconds and keeps trying.

Reply:
The docs say:

> An easy, but suboptimal way, to handle this is to..

Is there a more optimal way to handle this scenario?